transient/transient.py

```diff
@@ -46,6 +46,72 @@ def __needs_ssh_console(self) -> bool:
                 self.config.ssh_with_serial is True or
                 self.config.ssh_command is not None)
 
+    def __needs_to_copy_in_files_before_running(self) -> bool:
+        """Checks if at least one file or directory on the host needs to be copied into the VM
+           before starting the VM
+        """
+        return self.config.copy_in_before is not None
+
+    def __copy_in_files(self) -> None:
+        """Copies the given files or directories (located on the host) into the VM"""
+        path_mappings = self.config.copy_in_before
+        for path_mapping in path_mappings:
+            self.__copy_in(path_mapping)
+
+    def __copy_in(self, path_mapping: str) -> None:
+        """Copies the given file or directory (located on the host) into the VM"""
+        try:
+            host_path, vm_absolute_directory = path_mapping.split(':')
+        except ValueError:
+            raise RuntimeError(f'Invalid file mapping: {path_mapping}.' +
+                               ' -copy-in-before must be (path/on/host:/absolute/path/on/guest)')
+
+        if not os.path.exists(host_path):
+            raise RuntimeError(f'Host path does not exists: {host_path}')
+
+        if not vm_absolute_directory.startswith('/'):
+            raise RuntimeError(f'Absolute path for guest required: {vm_absolute_directory}')
+
+        env = os.environ
+        env['LIBGUESTFS_BACKEND'] = 'direct'
```

Comment:
Please add a comment explaining that this is to avoid a libvirt dependency from libguestfs